aws_lambda_powertools/event_handler/api_gateway.py

```diff
@@ -731,3 +731,84 @@ def register_route(func: Callable):
             self._routes[(rule, methods, cors, compress, cache_control)] = func
 
         return register_route
+
+
+class APIGatewayProxyEventResolver(ApiGatewayResolver):
+    current_event: APIGatewayProxyEvent
+
+    def __init__(
+        self,
+        cors: Optional[CORSConfig] = None,
+        debug: Optional[bool] = None,
+        serializer: Optional[Callable[[Dict], str]] = None,
+        strip_prefixes: Optional[List[str]] = None,
+    ):
+        """
+        Parameters
+        ----------
+        cors: CORSConfig
+            Optionally configure and enabled CORS. Not each route will need to have to cors=True
+        debug: Optional[bool]
+            Enables debug mode, by default False. Can be also be enabled by "POWERTOOLS_EVENT_HANDLER_DEBUG"
+            environment variable
+        serializer : Callable, optional
+            function to serialize `obj` to a JSON formatted `str`, by default json.dumps
+        strip_prefixes: List[str], optional
+            list of prefixes to be removed from the request path before doing the routing. This is often used
+            with api gateways with multiple custom mappings.
+        """
```

Comment:
```suggestion
```

These will be propagated from the superclass (granted, proxy_type as a helper will too but constructor will tell it's wrong) -- this helps us prevent multiple docstrings being accidentally outdated.

Tested on both VSCode and PyCharm (PyCharm is more accurate as it shows the constructor signature)

![image](https://user-images.githubusercontent.com/3340292/152007779-e6aa2c25-2fb6-484f-837b-596149dc8699.png)

![image](https://user-images.githubusercontent.com/3340292/152007917-ef91af79-04d4-4006-b19b-80b2b9c4b26f.png)